graphrag_sdk/ontology.py

```
@@ -103,12 +103,59 @@ def from_graph(graph: Graph):
+        # Extract attributes for each node label, limited by the specified sample size.
+        for lbls in n_labels:
+            l = lbls[0]
+            attributes = graph.query(
+                f"""MATCH (a:{l}) call {{ with a return [k in keys(a) | [k, typeof(a[k])]] as types }} 
+                WITH types limit {sample_size} unwind types as kt RETURN kt, count(1)""").result_set
+            ontology.add_entity(Entity(l, [Attribute(attr[0][0], attr[0][1]) for attr in attributes]))
+
+        # Extract attributes for each edge type, limited by the specified sample size.
+        for e_type in e_types:
+            attributes = graph.query(
+                            f"""MATCH ()-[a:{e_t}]->() call {{ with a return [k in keys(a) | [k, typeof(a[k])]] as types }} 
```

Ruff (0.8.2)

Undefined name `e_t`

(F821)